torchrec/distributed/planner/planners.py

```diff
@@ -302,3 +300,267 @@ def plan(
+        shards[0].rank = rank
+        return _get_parameter_sharding(
+            ShardingType.TABLE_WISE.value, shards, local_size, device_type
+        )
+
+    return _parameter_sharding_generator
+
+
+def row_wise_sharding() -> Callable[[nn.Parameter, int, int, str], ParameterSharding]:
+    def _parameter_sharding_generator(
+        param: nn.Parameter, local_size: int, world_size: int, device_type: str
+    ) -> ParameterSharding:
+        shards = _get_parameter_shards(
+            param,
+            ShardingType.ROW_WISE,
+            local_size,
+            world_size,
+        )
+        assert len(shards) <= world_size
+        for shard, rank in zip(shards, range(world_size)):
+            shard.rank = rank
+
+        return _get_parameter_sharding(
+            ShardingType.ROW_WISE.value, shards, local_size, device_type
+        )
+
+    return _parameter_sharding_generator
+
+
+def column_wise_sharding(
+    shard_dim: int = 128,
+    ranks: Optional[List[int]] = None,
+) -> Callable[[nn.Parameter, int, int, str], ParameterSharding]:
+    """
+    if ranks is not specified, shards will be placed in a uniform fashion.
+    If there are not enough shards, only the first ceil(embedding_dim/shard_dim) ranks will get
+    shards
+    """
+
+    def _parameter_sharding_generator(
+        param: nn.Parameter, local_size: int, world_size: int, device_type: str
+    ) -> ParameterSharding:
+        shards = _get_parameter_shards(
+            param, ShardingType.COLUMN_WISE, local_size, world_size, shard_dim
+        )
+
+        if ranks is None:
+            for shard, rank in zip(shards, range(world_size)):
+                shard.rank = rank
+        else:
+            assert len(shards) == len(ranks)
+            for shard, rank in zip(shards, ranks):
+                shard.rank = rank
+
+        return _get_parameter_sharding(
+            ShardingType.COLUMN_WISE.value, shards, local_size, device_type
+        )
+
+    return _parameter_sharding_generator
+
+
+def table_row_wise_sharding(
+    node_index: int,
+) -> Callable[[nn.Parameter, int, int, str], ParameterSharding]:
+    def _parameter_sharding_generator(
+        param: nn.Parameter, local_size: int, world_size: int, device_type: str
+    ) -> ParameterSharding:
+        shards = _get_parameter_shards(
+            param,
+            ShardingType.TABLE_ROW_WISE,
+            local_size,
+            world_size,
+        )
+
+        assert len(shards) <= local_size
+        for shard, rank in zip(shards, range(local_size)):
+            shard.rank = node_index + rank
+
+        return _get_parameter_sharding(
+            ShardingType.TABLE_ROW_WISE.value, shards, local_size, device_type
+        )
+
+    return _parameter_sharding_generator
+
+
+def construct_parameter_sharding_plans(
+    module: nn.Module,
+    per_param_sharding: Dict[
+        str, Callable[[nn.Parameter, int, int, str], ParameterSharding]
+    ],
+    sharder: ModuleSharder[nn.Module],
```

Comment:
We need to add a higher-level API, where we don't pass sharder. There is a module -> sharder dict. It should operate on top of it.